### ISSUE TITLE
Execution halted when running spectral matching (annotate_spectra.R)

### ISSUE THREAD

**Issue by RemyMG** [taxonomicallyinformedannotation/tima-r#69]
Hi Adriano,

I was running timaR smooth until the spectral matching step, where `snow` package seems to be absent and causes the workflow to stop. `timaR` package is well loaded, and I also manually loaded `snow` package, but with no success.

Any idea ? :)

Thanks for your help.

```
> library("timaR")
> ### Perform spectral_lib_matcher of my experimental spectra against ISDB 
> ### modify: inst/params/user/annotate_spectra.yaml with path to my experimental MGF
> source(file = "inst/scripts/annotate_spectra.R")
2023-07-12 16:59:19.624788 This script performs spectral similarity calculation to annotate against a spectral library. 
 
2023-07-12 16:59:19.625279 Authors:  AR 
 
2023-07-12 16:59:19.625644 Contributors: ... 
✔ skip target yaml_paths
✔ skip target paths
✔ skip target par_pre_par
✔ skip target par_def_pre_fea_tab
✔ skip target par_def_pre_tax
▶ start target lib_spe_is_lot_pos
File already exists. Skipping.
● built target lib_spe_is_lot_pos [0 seconds]
▶ start target lib_spe_is_lot_neg
File already exists. Skipping.
● built target lib_spe_is_lot_neg [0 seconds]
✔ skip target par_def_ann_spe
✔ skip target par_fin_par
✔ skip target lib_spe_is_lot_pre_pos
✔ skip target lib_spe_is_lot_pre_neg
✔ skip target par_usr_pre_fea_tab
✔ skip target par_usr_pre_tax
✔ skip target par_usr_ann_spe
✔ skip target par_pre_fea_tab
✔ skip target par_pre_tax
✔ skip target par_ann_spe
✔ skip target gnps_tables
✔ skip target gnps_spectra
✔ skip target input_spectra
▶ start target ann_spe_is_lot_pos
2023-07-12 16:59:26.792143 Loading spectra... 
2023-07-12 16:59:34.50321 Loading spectral library 
Error in loadNamespace("snow") : there is no package called 'snow'
Calls: local ... loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
```

**Comment by Adafede** [comment]
Hi @RemyMG, weird...are you using the latest version?

Could you attach your param file for the step?

**Comment by RemyMG** [comment]
Yes I'm using the latest version 2.8.1 , I cloned the git yesterday. You can download the data and parameters here [https://filesender.switch.ch/filesender2/?s=download&token=1a939897-5c80-4d9a-8e0e-2c7718077bba]

Let me know if I'm doing well

**Comment by Adafede** [comment]
It is weird since `snow` is called for parallel computing on Windows (should be installed by default), but the default parameters should not use parallelization...

In all my last tests, `snow` did install correctly (see https://github.com/taxonomicallyinformedannotation/tima-r/actions/runs/5389085875/jobs/9782574028#step:4:431 for example)...maybe retry a clean isntall?

**Comment by RemyMG** [comment]
I tried with a clean install and I get the same issue. I also get a similar message when I try to run the GUI:

```
(base) PS C:\Users\marcellr\GitHub\tima-r> Rscript inst/scripts/run_app.R
Error in loadNamespace(x) : there is no package called 'shiny'
Calls: loadNamespace -> withRestarts -> withOneRestart -> doWithOneRestart
Execution halted
```

And of course `shiny` is well installed

**Comment by RemyMG** [comment]
Hi @Adafede, 

Since the latest update 2.8.2, spectral matching `annotate_spectra.R` works fine now, but the step just after `prepare_annotations_spectra.R` comes with an issue, a file is probably missing: 

```
> source(file = "inst/scripts/prepare_annotations_spectra.R")
2023-07-13 10:52:51.170467 This script formats ISDB results 
2023-07-13 10:52:51.17114 Authors:  AR 
 
2023-07-13 10:52:51.171698 Contributors: ... 
✔ skip target yaml_paths
✔ skip target paths
✔ skip target par_def_pre_lib_sop_clo
✔ skip target par_pre_par
✔ skip target par_def_pre_fea_tab
▶ start target lib_sop_lot
A file with the same size is already present. Skipping
● built target lib_sop_lot [0.28 seconds]
✔ skip target par_def_pre_lib_sop_lot
✔ skip target par_def_pre_lib_sop_mer
✔ skip target par_def_pre_ann_spe
✔ skip target par_def_pre_tax
✔ skip target par_def_pre_lib_spe
✔ skip target lib_sop_ecm
✔ skip target par_def_pre_ann_gnp
✔ skip target par_def_pre_lib_sop_ecm
▶ start target lib_spe_is_lot_pos
File already exists. Skipping.
● built target lib_spe_is_lot_pos [0 seconds]
▶ start target lib_spe_is_lot_neg
File already exists. Skipping.
● built target lib_spe_is_lot_neg [0.02 seconds]
✔ skip target par_def_ann_spe
✔ skip target par_fin_par
✔ skip target lib_spe_is_lot_pre_pos
✔ skip target lib_spe_is_lot_pre_neg
✔ skip target par_usr_pre_lib_sop_clo
✔ skip target par_usr_pre_fea_tab
✔ skip target par_usr_pre_tax
✔ skip target par_usr_pre_lib_sop_lot
✔ skip target par_usr_pre_lib_sop_mer
✔ skip target par_usr_pre_ann_spe
✔ skip target par_usr_pre_lib_spe
✔ skip target par_usr_pre_ann_gnp
✔ skip target par_usr_pre_lib_sop_ecm
✔ skip target par_usr_ann_spe
✔ skip target par_pre_lib_sop_clo
✔ skip target par_pre_fea_tab
✔ skip target par_pre_tax
✔ skip target par_pre_lib_sop_lot
✔ skip target par_pre_lib_sop_mer
✔ skip target par_pre_ann_spe
✔ skip target par_pre_lib_spe
✔ skip target par_pre_ann_gnp
✔ skip target par_pre_lib_sop_ecm
✔ skip target par_ann_spe
✔ skip target lib_sop_clo_pre
✔ skip target lib_sop_lot_pre
▶ start target lib_spe_exp_int_raw
✖ error target lib_spe_exp_int_raw
▶ end pipeline [3.07 seconds]
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: _paths_ missing files: data/source/libraries/spectra/exp/spectral_lib_mini_with_rt.mgf
```

**Comment by Adafede** [comment]
Oh, this one is a nice catch!

It was trying to perform spectral matching against an internal library (the one of Wolfender lab btw) by default. It is advanced use and was not supposed to be done by default.

I just pushed a small fix (14d025994de3fcd4ac05edfde566efb10615dd87), tell me if everything works fine now! 😊 

P.S.: Only a small file to download to unlock the internal library match in case 😜

**Comment by RemyMG** [comment]
That works perfectly fine now ! From spectral matching to metabolites reranking. Thanks a lot @Adafede !!
